click_spinner/__init__.py

```diff
@@ -1,19 +1,54 @@
 import sys
 import threading
 import itertools
+from ._version import get_versions
+
+_iter_vals = ["-", "/", "|", "\\"]
 
 
 class Spinner(object):
-    spinner_cycle = itertools.cycle(['-', '/', '|', '\\'])
+    spinner_iter = _iter_vals
+
+    def __init__(
+        self,
+        beep=False,
+        disable=False,
+        force=False,
+        stream=sys.stdout,
+        direction="counter-clockwise",
+    ):
+        if not isinstance(beep, bool):
+            raise TypeError("must be bool, not %s" % (type(beep).__name__))
```

Comment:
This seems unnecessary: there’s no reason `beep` must be a boolean, it just has to act as such.